adam-python/setup.py

```diff
@@ -16,15 +16,119 @@
 # limitations under the License.
 #
 
+from __future__ import print_function
+import glob
+import os
+import sys
 from setuptools import find_packages, setup
+
 from version import version as adam_version
 
-setup(
-    name='bdgenomics.adam',
-    version=adam_version,
-    description='A fast, scalable genome analysis system',
-    author='Frank Austin Nothaft',
-    author_email='fnothaft@berkeley.edu',
-    url="https://github.com/bdgenomics/adam",
-    install_requires=[],
-    packages=find_packages(exclude=['*.test.*']))
+if sys.version_info < (2, 7):
+    print("Python versions prior to 2.7 are not supported for pip installed ADAM.",
+          file=sys.stderr)
+    exit(-1)
+
+# Provide guidance about how to use setup.py
+incorrect_invocation_message = """
+If you are installing PyADAM from ADAM's source, you must first build ADAM and
+run make develop.
+
+    To build ADAM with maven you can run:
+      ./build/mvn -DskipTests clean package
+    Building the source dist is done in the Python directory:
+      cd adam-python
+      make sdist
+      pip install dist/*.tar.gz"""
+
+# Figure out where the jars are we need to package
+ADAM_HOME = os.path.abspath("../")
+TEMP_PATH = "deps"
+
+JARS_PATH = glob.glob(os.path.join(ADAM_HOME, "adam-assembly/target/adam-assembly*.jar"))
```

Comment:
Do we need the same regex mess to avoid -sources and -javadoc jars as in `find-adam-assembly.sh` here?